lib/Parser/Parse.cpp

```diff
@@ -8655,6 +8655,12 @@ ParseNodePtr Parser::ParseVariableDeclaration(
             nameHintLength = pid->Cch();
             nameHintOffset = 0;
 
+            if ((declarationType == tkCONST || declarationType == tkLET) &&
+                wcsncmp(pNameHint, L"let", 4) == 0)
```

Comment:
IdentPtrs (aka PIDs) are interned strings which means you need only do a pointer comparison with the PID you already know to be `let`.  No need to do a string comparison.

In this case you want to do `pid == wellKnownPropertyPids.let`.

`wellKnownPropertyPids` is a member of `Parser`.